src/Components/Common/TemperatureSelector.tsx

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from "react";
+import { classNames } from "../../Utils/utils";
+import { SelectFormField } from "../Form/FormFields/SelectFormField";
+
+export const TemperatureSelector = (props: any) => {
+  const getTemperaturePreferance = () => {
+    if (window && window.localStorage) {
+      return localStorage.getItem("temperature");
+    }
+  };
```

Comment:
Move this to `Utils.tsx` or `utils/DevicePreferences.tsx` and add types for the return type. Maybe store it and type it as `"C" | "F"` instead of full text, this way we can easily add it to translation files too.